### ISSUE TITLE
On fetch failure the previous result is discarted

### ISSUE THREAD

**Issue by berlincount** [deg0nz/MMM-PublicTransportBerlin#31]
I keep seeing fetch failures (400), while the data from the previous fetch would still be current enough to display _something_ ..

Maybe keeping the last result until the next successful fetch would make sense?

**Comment by deg0nz** [comment]
Yes, this could make sense, but I think we would have to specify a time for which the old data is still "fresh enough" to be kept.
So we need some kind timeout time for the old data until we finally show the fetcher errors and remove the old data...

**Comment by berlincount** [comment]
Well, marking it stale can be done immediately either way .. no upcoming trains known anymore is the ultimate error state ..

**Comment by deg0nz** [comment]
I want to have this feature and I'm going to implement this at some time... Summer and the upcoming festival season cuts time for me.

So any contribution is welcome! Please feel free to make PR. :)

I'm adding the "Help wanted" label now..

**Comment by deg0nz** [comment]
This could be nicely implemented with [`cached-hafas-client`](https://github.com/derhuerst/cached-hafas-client). This way, we could save some more data for future departures if needed...

Let's see, when the sqlite version is out and check if this is not overkill...

**Comment by deg0nz** [comment]
So, a small update on that: [`cached-hafas-client`](https://github.com/derhuerst/cached-hafas-client)  seems to be perfect for the job without much effort for implementing this feature.

Although I would like to wait until the SQLite datastore is implemented, so we don't have to bother MagicMirror-users to install a Redis server on their machines. With SQLite we will probably be able to automate all this.

**Comment by berlincount** [comment]
seems like the sqlite backend is available now! :)
